typescript/infra/scripts/core.ts

```diff
@@ -17,23 +18,32 @@ async function main() {
   const multiProvider = await config.getMultiProvider();
   const deployer = new AbacusCoreInfraDeployer(multiProvider, config.core);
 
-  const contracts = await deployer.deploy();
-
-  writeJSON(
-    getCoreContractsSdkFilepath(),
-    `${environment}.json`,
-    serializeContracts(contracts),
-  );
-  writeJSON(
-    getCoreVerificationDirectory(environment),
-    'verification.json',
-    deployer.verificationInputs,
-  );
-  deployer.writeRustConfigs(
-    environment,
-    getCoreRustDirectory(environment),
-    contracts,
-  );
+  try {
+    const contracts = await deployer.deploy();
+    writeJSON(
+      getCoreContractsSdkFilepath(),
+      `${environment}.json`,
+      serializeContracts(contracts),
+    );
+    writeJSON(
+      getCoreVerificationDirectory(environment),
+      'verification.json',
+      deployer.verificationInputs,
+    );
+    deployer.writeRustConfigs(
+      environment,
+      getCoreRustDirectory(environment),
+      contracts,
+    );
+  } catch (e) {
+    console.error(e);
+    // persist partial deployment
+    writeJSON(
+      getEnvironmentDirectory(environment),
+      'partial_core_addresses.json',
```

Comment:
Where is this read? Not seeing this file name referenced anywhere else

Reply:
it's not, in failure mode would have to read it in